pkg/client/request_test.go

```diff
@@ -736,7 +734,9 @@ func TestDoRequestNewWay(t *testing.T) {
 	} else if !api.Semantic.DeepDerivative(expectedObj, obj) {
 		t.Errorf("Expected: %#v, got %#v", expectedObj, obj)
 	}
-	fakeHandler.ValidateRequest(t, "/api/v1beta2/foo/bar/baz?timeout=1s", "POST", &reqBody)
+	requestURL := testapi.ResourcePathWithPrefix("foo/bar", "", "", "baz")
```

Comment:
Renamed the function to ResourcePathWithPrefixAndNamespaceQuery to remove confusion.